src/main/java/co/cask/http/HttpMethodInfo.java

```diff
@@ -114,7 +115,13 @@ private void bodyConsumerChunk(ChannelBuffer buffer) throws HandlerException {
     try {
       bodyConsumer.chunk(buffer, responder);
     } catch (Throwable t) {
-      throw bodyConsumerError(t);
+      try {
+        bodyConsumerError(t);
+      } catch (Throwable t2) {
+        exceptionHandler.handle(t2, request, responder);
```

Comment:
It would be a good idea to log the first exception, so that we don't lose it in this case.